<?xml version="1.0"?>

<!--
# Launch file for the base components of the Navigation stack.
# Author: John Lingi
# Date: 02/18
-->

<launch>

    <!--Base_control_sig -->
    <node name="vel_topic_converter" pkg="navigation" type="vel_topic_converter.py" output="screen"/>

    <!-- Odometry -->
    <node name="odometry_publisher_cmd_vel" pkg="navigation" type="odometry_publisher_cmd_vel.py" output="screen"/>
    <!-- NEW: python odom publisher that uses key frame laser scan matching -->

    <!-- Load map -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find navigation)/maps/huxley_311_2.yaml"/>

    <!-- Hector mapping for localisation -->
    <node name="hector_mapping" pkg="hector_mapping" type="hector_mapping" output="screen">

        <!-- remap topic publishing to, to avoid conflict with map_server -->
        <remap from="map" to="hector_map"/>
        <param name="map_frame" value="map" />
        <param name="base_frame" value="base_link" />
        <param name="odom_frame" value="odom" />
    </node>

    <!-- Motion planning -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base">

        <rosparam file="$(find navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation)/config/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation)/config/teb_local_planner_params.yaml" command="load" />

        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
        <param name="controller_frequency" value="40.0" />
        <param name="controller_patience" value="10.0" />
        <param name="clearing_rotation_allowed" value="false" />

    </node>

    <!-- Movement base controller for mbed -->
    <node pkg="navigation" type="movement_control.py" name="base_motion_control" output="screen"/>

    <!-- Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigation)/rviz/deniro_nav.rviz"/>

</launch>
